src/ripple/nodestore/DatabaseShard.h

```diff
@@ -61,7 +61,7 @@ class DatabaseShard : public Database
 
         @return `true` if the database initialized without error
     */
-    virtual bool
+    virtual [[nodiscard]] bool
```

Comment:
Clang is unhappy with the `[[nodiscard]]` after the `virtual`.  However
```
[[nodiscard]] virtual bool
```
works.  I suggest changing this one and the other four `[[nodiscard]]`s in this file.